src/components/tooltip/tooltip.js

```diff
@@ -0,0 +1,191 @@
+
+class Tooltip extends React.Component {
+  static displayName = 'ToolTip';
+
+  state = {
+    open: this.isControlled ? this.props.open : false,
+  };
+
+  static getDerivedStateFromProps = (props, state) => ({
+    id: getFieldId(props, state, sequentialId),
+  });
+
+  componentDidMount() {
+    // eslint-disable-next-line no-console
+    const childrenProps = this.props.children.props;
+
+    invariant(
+      !(childrenProps.disabled || childrenProps.isDisabled),
+      [
+        'ui-kit: you are providing a disabled `button` child to the Tooltip component.',
+        'A disabled element does not fire events.',
+        "Tooltip needs to listen to the child element's events to display the title.",
+        '',
+        'Place a `div` container on top of the element.',
+      ].join('\n')
+    );
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.leaveTimer);
+  }
+
+  handleEnter = event => {
+    const childrenProps = this.props.children.props;
+
+    // Remove the title ahead of time.
+    // We don't want to wait for the next render commit.
+    // We would risk displaying two tooltips at the same time (native + this one).
+    this.childrenRef.setAttribute('title', '');
+
+    if (event.type === 'mouseover' && childrenProps.onMouseOver) {
+      childrenProps.onMouseOver(event);
+    }
+
+    if (event.type === 'focus' && childrenProps.onFocus) {
+      childrenProps.onFocus(event);
+    }
+
+    if (!this.state.open && !this.isControlled) {
+      this.setState({
+        open: true,
+      });
+    }
+
+    if (this.props.onOpen) {
+      this.props.onOpen(event);
+    }
+  };
+
+  handleLeave = event => {
+    const childrenProps = this.props.children.props;
+    clearTimeout(this.leaveTimer);
+
+    if (event.type === 'mouseleave' && childrenProps.onMouseLeave) {
+      childrenProps.onMouseLeave(event);
+    }
+
+    if (event.type === 'blur' && childrenProps.onBlur) {
+      childrenProps.onBlur(event);
+    }
+
+    if (this.props.leaveDelay) {
+      this.leaveTimer = setTimeout(() => {
+        this.handleClose();
+      }, this.props.leaveDelay);
+    } else {
+      this.handleClose();
+    }
+  };
+
+  handleClose = event => {
+    if (!this.isControlled) {
+      this.setState({
+        open: false,
+      });
+    }
+    if (this.props.onClose) {
+      this.props.onClose(event);
+    }
+  };
+
+  onRootRef = ref => {
+    this.childrenRef = ref;
+  };
+
+  isControlled = this.props.open != null;
+
+  render() {
+    const open = this.isControlled ? this.props.open : this.state.open;
+
+    const childrenProps = {
+      'aria-describedby': open ? this.state.id : null,
+      // for seo and accessibility, we add the tooltip's title
+      // as a native title when the title is hidden
+      title:
+        !open && typeof this.props.title === 'string' ? this.props.title : null,
+      ...this.props.children.props,
+    };
+
+    childrenProps.onMouseOver = this.handleEnter;
```

Comment:
Why do we do this here and not in L117? Just `onMouseLeave: this.handleLeave` etc?